src/components/KillCam.tsx

```diff
@@ -1,49 +1,85 @@
-import React, {useEffect, useState, useRef} from 'react';
+import React, { useEffect, useState, useRef, useContext } from 'react';
 import { useNavigate } from 'react-router-dom';
 import {
   FaceMatcher,
   createCanvasFromMedia,
   matchDimensions,
   detectAllFaces,
   resizeResults,
-  draw } from 'face-api.js';
+  draw
+} from 'face-api.js';
 import { useWebcam } from '../contexts/WebcamProvider';
 import TargetRecognition from './KillProgress';
+import { useAuth0 } from "@auth0/auth0-react";
+
+import SocketContext from '../contexts/Socket/SocketContext';
+
 
 type KillCamProps = {
   faceMatcher: (FaceMatcher | null)
 }
 
-const KillCam: React.FC<KillCamProps> = ({faceMatcher}) => {
+const KillCam: React.FC<KillCamProps> = ({ faceMatcher }) => {
   const webcamContext = useWebcam();
   const webcamRef = webcamContext?.webcamRef;
-  const  videoStarted = webcamContext?.videoStarted;
+  const videoStarted = webcamContext?.videoStarted;
   const navigate = useNavigate();
   const videoHeight = window.innerHeight;
   const videoWidth = window.innerWidth;
   const displaySize = { width: videoWidth, height: videoHeight };
   const canvasRef = useRef<HTMLCanvasElement | null>(null);
   let [targetCounter, setTargetCounter] = useState(0);
   let wasBountyDetected = false;
-  
+  const { user } = useAuth0();
+
+  // socket contexts
+  const { games, users } = useContext(SocketContext).SocketState;
+  const { UpdateGameStatus } = useContext(SocketContext);
+
+  // storing username of hunted so tensor can compare
+  const [huntedUsername, setHuntedUsername] = useState<string>('');
+
+
+  // creating the canvas when the component mounts
   useEffect(() => {
     createCanvas()
   }, [])
 
+  // if faceMatcher and huntedUsername are both set, start the handleVideoPlay()
   useEffect(() => {
-    if (faceMatcher) {
+    if (faceMatcher && huntedUsername) {
       handleVideoOnPlay();
       const intervalId = setInterval(updateCounter, 500);
       return () => clearInterval(intervalId);
     }
-  }, [faceMatcher])
+  }, [faceMatcher, huntedUsername])
 
-  useEffect(()=> {
-    console.log('targetCount', targetCounter)
+  // whenever targetCounter is updated, if it's at 10, navigate the users
+  useEffect(() => {
+    if (targetCounter === 5) {
```

Comment:
if the target counter is set to 5, it means that the bounty only has to be correctly tracked for 2.5 seconds. I think we should change it back to ten because 2.5 feels very small. when we start actually play testing we'll get a feel for what feels like a good number.